feat: Slack AI & Assistants Compatibility

MattDavisRV opened the pull request:
Added new function for Slack AI & Assistants, information can be found here: https://api.slack.com/docs/apps/ai

New Events:
https://api.slack.com/events/assistant_thread_started
https://api.slack.com/events/assistant_thread_context_changed

New API Calls:
https://api.slack.com/methods/assistant.threads.setStatus
https://api.slack.com/methods/assistant.threads.setSuggestedPrompts
https://api.slack.com/methods/assistant.threads.setTitle

Tests have been added and pass all checks via `make pr-prep`.

MattDavisRV committed:
Adding compatibility for Slack Assistants

MattDavisRV committed:
Added Assistant Tests

MattDavisRV committed:
Added event tests for Assistant

MattDavisRV commented:
This PR resolves [issue 1326](https://github.com/slack-go/slack/issues/1326).

MattDavisRV committed:
Updated test handler to return standard 'ok' response

lorenzoaiello commented on assistant.go:
```diff
@@ -0,0 +1,160 @@
+}
+
+// AssistantThreadSetSuggestedPromptsParameters are the parameters for AssistantThreadSetSuggestedPrompts
+type AssistantThreadsSetSuggestedPromptsParameters struct {
```
Could you please add the `title` optional argument? I know the [documentation](https://api.slack.com/methods/assistant.threads.setSuggestedPrompts#arg_title) says its currently ignored by the Slack client, but I'd like for this to match the API spec.

lorenzoaiello commented on assistant.go:
```diff
@@ -0,0 +1,160 @@
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
```
I don't think this would ever happen since the fields are required both in the API call and the input struct - can you confirm?

lorenzoaiello commented on assistant.go:
```diff
@@ -0,0 +1,160 @@
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
```
I don't think this would ever happen since the fields are required both in the API call and the input struct - can you confirm?

lorenzoaiello commented on assistant.go:
```diff
@@ -0,0 +1,160 @@
+		"token": {api.token},
+	}
+
+	if params.ChannelID != "" {
```
I don't think this would ever happen since the fields are required both in the API call and the input struct - can you confirm?

lorenzoaiello reviewed (commented):
Hey @MattDavisRV , thanks for the PR! I've got some minor feedback items I added.

It also looks like you may have included some new events, which while very much appreciated, may not be related to the Assistant Events.

Could you please submit these changes in a different PR so we can keep PRs scoped to a single issue or share the documentation where they are required for the Assistant functionality?

MattDavisRV committed:
Removed check for channel_id before making request, added Title as param, not currently used by Slack

MattDavisRV committed:
Removed events that are not part of this PR, unsure how they were added initially

MattDavisRV committed:
Removed additional events and properties not part of this PR

MattDavisRV committed:
Added Section Block Option for Expand

MattDavisRV commented:
@lorenzoaiello, sorry about the initially cluttered PR, I didn't look at the changed files and I'm unsure how all of the additional changes were added in.  I've cleaned up the PR to only be my changes.

I've also added in an additional change for the Section Block that was added due to the AI changes.  I was going to submit an additional PR with it, but I decided to add it in here.

lorenzoaiello commented:
@MattDavisRV , can you please resolve the conflicts?

MattDavisRV committed:
Merge branch 'master' into slack-ai-assistant

MattDavisRV committed:
Refactored tests

MattDavisRV committed:
Merge branch 'slack-ai-assistant' of github.com:MattDavisRV/slack into slack-ai-assistant

MattDavisRV committed:
Updated tests due to form variables

MattDavisRV committed:
go fmt

MattDavisRV commented:
@lorenzoaiello all set, I refactored my tests, it looks like i can't use a handler function with form variables, I noticed other tests were using the standard "ok handler", so I switched to that.  All tests are passing now.

MattDavisRV commented:
Hey @lorenzoaiello, anything else I need to do here to get this merged in?
